src/app/auth/auth-registration/auth-registration.component.ts

```diff
@@ -0,0 +1,259 @@
+                false
+              );
+            }
+          );
+      } else {
+        this.utils.popUp(
+          "shortMessage",
+          {
+            message: "Registration link invalid"
+          },
+          false
+        );
+      }
+    });
+  }
+
+  private autoGeneratePassword() {
+    let text = Md5.hashStr("").toString();
+    let autoPass = text.substr(0, 8);
+    return autoPass;
+  }
+
+  openLink() {
+    window.open(
+      "https://images.practera.com/terms_and_conditions/practera_default_terms_conditions_july2018.pdf",
+      "_system"
+    );
+  }
+
+  register() {
+    this.validateRegistration();
+    if ((this.registerationForm.valid) && (this.errors.length === 0)) {
+      this.authService
+        .saveRegistration({
+          password: this.confirmPassword
+        })
+        .subscribe(
+          response => {
+            this.authService
+              .login({
+                email: this.user.email,
+                password: this.password
+              })
+              .subscribe(
+                response => {
+                  this.authService.me().subscribe(
+                    response => {
+                      this.utils
+                        .popUp(
+                          "shortMessage",
+                          {
+                            message: "Registration success"
+                          },
+                          false
+                        )
+                        .then(() => {});
+                    },
+                    error => {}
+                  );
+                },
+                error => {}
+              );
+          },
+          error => {}
+        );
+    }
+  }
+
+  removeErrorMessages() {
+    this.errors = [];
+  }
+
+  validateRegistration() {
+    this.errors = [];
+    if (this.registerationForm.valid) {
+      let pass = this.registerationForm.controls.password.value;
+      let confirmPass = this.registerationForm.controls.confirmPassword.value;
+      if (pass !== confirmPass) {
+        this.errors.push("Your passwords don't match.");
+      } else if (!this.isAgreed) {
+        this.errors.push("You need to agree with terms and Conditions.");
+      }
+    } else {
+      for (let conrtoller in this.registerationForm.controls) {
+        for (let key in this.registerationForm.controls[conrtoller].errors) {
+          if (this.registerationForm.controls[conrtoller].errors.hasOwnProperty(key)) {
+            switch (key) {
+              case "required":
+                this.errors.push("Fill required field");
+                break;
+              case "minlength":
+                this.errors.push(
+                  "Your password needs to be more than 8 characters."
+                );
+                break;
+              default:
+              this.errors.push(this.registerationForm.controls.errors[key]);
+            }
+            return;
+          }
+        }
+      }
+    }
+    // if (check) {
```

Comment:
What are the codes below? If we need to keep them, put some comment about the reason, otherwise remove them